src/System.Private.CoreLib/src/System/Attribute.CoreCLR.cs

```diff
@@ -569,7 +569,7 @@ public static Attribute[] GetCustomAttributes(ParameterInfo element, Type attrib
             if (member.MemberType == MemberTypes.Method && inherit)
                 return InternalParamGetCustomAttributes(element, attributeType, inherit);
 
-            return (Attribute[])element.GetCustomAttributes(attributeType, inherit);
+            return (element.GetCustomAttributes(attributeType, inherit) as Attribute[])!;
```

Comment:
@buyaa-n considering the test failed I'd presume the function should return `Attribute[]?` Is the test testing super edge case or some broken scenario?

Reply:
Tests expecting RuntimeBinderException  so it might edge case, before with explicit casting it was throwing InvalidCastException instead RuntimeBinderException.
Not sure about if we should make return type nullable, I don't see much usage of this method in CoreClr, similar methods in MemberInfo returns non null and accessed without null check a lot ...

Reply:
Please revert the change and suppress the warning with !. Same for everywhere else the as cast was changed to an explicit cast. Thanks.

Reply:
@stephentoub it is reverted from explicit cast, will check same for other files